src/common/pf_alarm.c

```diff
@@ -704,53 +704,64 @@ static int pf_alarm_apmr_frame_handler (
    pf_apmx_t * p_apmx = (pf_apmx_t *)p_arg;
    pf_apmr_msg_t * p_apmr_msg;
    uint16_t nbr;
-   int ret = 0; /* Failed to handle frame. The calling function needs to free
+   int ret = 0; /* Failed to handle frame. The frame handler needs to free
                    the buffer. */
 
-   LOG_INFO (
-      PF_ALARM_LOG,
-      "Alarm(%d): Received %s prio alarm frame.\n",
-      __LINE__,
-      p_apmx->high_priority ? "high" : "low");
    if (p_buf != NULL)
    {
-      nbr = p_apmx->apmr_msg_nbr++; /* ToDo: Make atomic */
-      if (p_apmx->apmr_msg_nbr >= NELEMENTS (p_apmx->apmr_msg))
-      {
-         p_apmx->apmr_msg_nbr = 0;
-      }
-      p_apmr_msg = &p_apmx->apmr_msg[nbr];
-      p_apmr_msg->p_buf = p_buf;
-      p_apmr_msg->frame_id_pos = frame_id_pos;
       if (p_apmx->p_alarm_q != NULL)
       {
+         nbr = p_apmx->apmr_msg_nbr++; /* ToDo: Make atomic */
```

Comment:
Remove ToDo?
Can p_apmx->apmr_msg_nbr be updated when post succeeds instead?

Reply:
Now updated when post succeeds
Kept TODO
Done